launch/tier4_map_launch/launch/map.launch.py

```diff
@@ -114,42 +114,56 @@ def generate_launch_description():
+            map_hash_generator,
+        ]
+    )
+
+    return [group]
+
+
+def generate_launch_description():
+    launch_arguments = []
+
     def add_launch_arg(name: str, default_value=None, description=None):
-        return DeclareLaunchArgument(name, default_value=default_value, description=description)
+        launch_arguments.append(
+            DeclareLaunchArgument(name, default_value=default_value, description=description)
+        )
+
+    add_launch_arg("map_path", "", "path to map directory"),
+    add_launch_arg(
+        "lanelet2_map_path",
+        [LaunchConfiguration("map_path"), "/lanelet2_map.osm"],
+        "path to lanelet2 map file",
+    ),
+    add_launch_arg(
+        "pointcloud_map_path",
+        [LaunchConfiguration("map_path"), "/pointcloud_map.pcd"],
+        "path to pointcloud map file",
+    ),
+    add_launch_arg("use_intra_process", "false", "use ROS2 component container communication"),
+    add_launch_arg("use_multithread", "false", "use multithread"),
+
+    set_container_executable = SetLaunchConfiguration(
+        "container_executable",
+        "component_container",
+        condition=UnlessCondition(LaunchConfiguration("use_multithread")),
+    )
+
+    set_container_mt_executable = SetLaunchConfiguration(
+        "container_executable",
+        "component_container_mt",
+        condition=IfCondition(LaunchConfiguration("use_multithread")),
+    )
 
     return launch.LaunchDescription(
-        [
-            add_launch_arg("map_path", "", "path to map directory"),
-            add_launch_arg(
-                "lanelet2_map_path",
-                [LaunchConfiguration("map_path"), "/lanelet2_map.osm"],
-                "path to lanelet2 map file",
-            ),
-            add_launch_arg(
-                "pointcloud_map_path",
-                [LaunchConfiguration("map_path"), "/pointcloud_map.pcd"],
-                "path to pointcloud map file",
-            ),
-            add_launch_arg(
-                "use_intra_process", "false", "use ROS2 component container communication"
-            ),
-            add_launch_arg("use_multithread", "false", "use multithread"),
-            SetLaunchConfiguration(
-                "container_executable",
-                "component_container",
-                condition=UnlessCondition(LaunchConfiguration("use_multithread")),
-            ),
-            SetLaunchConfiguration(
-                "container_executable",
-                "component_container_mt",
-                condition=IfCondition(LaunchConfiguration("use_multithread")),
-            ),
-            GroupAction(
-                [
-                    PushRosNamespace("map"),
-                    container,
-                    map_hash_generator,
-                ]
-            ),
+        launch_arguments
+        + [
+            set_container_executable,
+            set_container_mt_executable,
         ]
+        + [OpaqueFunction(function=launch_setup)]
```

Comment:
What for did you have to use `OpaqueFunction`?

Reply:
This is used in the below file.
https://github.com/autowarefoundation/autoware.universe/blob/main/launch/tier4_control_launch/launch/control.launch.py#L385
Actually I copied it.

Reply:
The file uses `context`, but this file doesn't.
So, could you remove it from this PR? :pray:

Reply:
```suggestion
        + [launch_setup]
```
This way?

Reply:
No. Please remove `launch_setup` itself.

Reply:
You mean I have to delete `launch_setup` function ?

Reply:
Exactly. It's not necessary because it doesn't use the `context` variable.

Reply:
I split https://github.com/autowarefoundation/autoware.universe/pull/952 into two pull requests, refactor and feature.
Now this pull request is refactor. So, I'm going to use `context` variables after merging this to make feature pull request.

Reply:
I see. In that case, it's acceptable.